### Eclipse Workspace Patch 1.0
#P laneweb
Index: src/main/webapp/resources/test/search-indicator/search-indicator-test.html
===================================================================
--- src/main/webapp/resources/test/search-indicator/search-indicator-test.html	(revision 104454)
+++ src/main/webapp/resources/test/search-indicator/search-indicator-test.html	(working copy)
@@ -4,10 +4,10 @@
         <title>lane-search-indicator javascript unit tests</title>
     </head>
     <body>
+        <div id="searchIndicator"></div>
         <script type="text/javascript" src="../../javascript/yui-3.6.0.js"></script>
         <script src="../../javascript/lane.js" type="text/javascript"></script>
-        <script src="../../javascript/model.js" type="text/javascript"></script>
-        <script src="../../javascript/lane-search-indicator.js" type="text/javascript"></script>
+        <script src="../../javascript/search-indicator.js" type="text/javascript"></script>
         <script src="search-indicator-test.js" type="text/javascript"></script>
     </body>
 </html>
Index: src/main/webapp/resources/javascript/lane-search.js
===================================================================
--- src/main/webapp/resources/javascript/lane-search.js	(revision 104454)
+++ src/main/webapp/resources/javascript/lane-search.js	(working copy)
@@ -6,7 +6,7 @@
             searchOptions = searchSourceSelect.all('option'),
             searchTipsLink = Y.one('#searchTips a'),
             selectedOption = searchOptions.item(searchSourceSelect.get('selectedIndex')),
-            searchIndicator = new LANE.SearchIndicator(),
+            searchIndicator = Y.lane.SearchIndicator,
             searchTextInput = new Y.lane.TextInput(form.one('#searchTerms')),
             getLimitForSource = function(source) {
                 var limit = "";
Index: src/main/webapp/resources/templates/template.xml
===================================================================
--- src/main/webapp/resources/templates/template.xml	(revision 104454)
+++ src/main/webapp/resources/templates/template.xml	(working copy)
@@ -45,8 +45,7 @@
                                         >Textbooks Search</option>
                                 </select>
                                 <span id="searchTips"><a href="/help/choose-search.html">choose your search</a> &#187; </span>
-                                <div id="searchIndicator"><img alt="[searching]"
-                                    src="/resources/images/search-indicator.gif"/></div>
+                                <div id="searchIndicator"></div>
                                 <input name="q" id="searchTerms" type="text"/>
                                 <input id="searchSubmit" type="submit" value="search"/>
                                 <a id="searchReset" title="Clear Search" href="/">×</a>
Index: src/main/webapp/resources/test/search-indicator/search-indicator-test.js
===================================================================
--- src/main/webapp/resources/test/search-indicator/search-indicator-test.js	(revision 104454)
+++ src/main/webapp/resources/test/search-indicator/search-indicator-test.js	(working copy)
@@ -6,36 +6,15 @@
 
     var searchIndicatorTestCase = new Y.Test.Case({
         name: "Lane Search Indicator Test Case",
-        setUp: function() {
-            var indicator = Y.one("#searchIndicator");
-            if (indicator) {
-                indicator.remove();
-            }
-        },
-        testImageNotPresent: function() {
-            Y.Assert.isNull(Y.one("#searchIndicator"));
-        },
-        testConstructorCreatesImage: function() {
-            new LANE.SearchIndicator();
-            Y.Assert.isObject(Y.one("#searchIndicator"));
-        },
+        
+        indicator : Y.lane.SearchIndicator,
+        
         testShowAndHide: function() {
-            var indicator = new LANE.SearchIndicator();
             var node = Y.one("#searchIndicator");
-            indicator.show();
-            Y.Assert.areEqual("block", node.getStyle("display"));
-            indicator.hide();
-            Y.Assert.areEqual("none", node.getStyle("display"));
-        },
-        testTwoIndicatorsPlayNice: function() {
-            var indicator1 = new LANE.SearchIndicator();
-            var indicator2 = new LANE.SearchIndicator();
-            Y.Assert.areEqual(1, Y.all("#searchIndicator").size());
-            var node = Y.one("#searchIndicator");
-            indicator1.show();
-            Y.Assert.areEqual("block", node.getStyle("display"));
-            indicator2.hide();
-            Y.Assert.areEqual("none", node.getStyle("display"));
+            this.indicator.show();
+            Y.Assert.isTrue(node.hasClass("on"));
+            this.indicator.hide();
+            Y.Assert.isFalse(node.hasClass("on"));
         }
     });
     
Index: src/main/webapp/resources/css/search-form.css
===================================================================
--- src/main/webapp/resources/css/search-form.css	(revision 104454)
+++ src/main/webapp/resources/css/search-form.css	(working copy)
@@ -89,11 +89,17 @@
 }
 #searchIndicator {
     display : none;
+    background-image: url(/././resources/images/search-indicator.gif);
     position : absolute;
     top : 0px;
     left : 280px;
+    width: 220px;
+    height: 19px;
     z-index : 1;
 }
+#searchIndicator.on {
+	display:block
+}
 #searchTips {
     position : absolute;
     top : 3px;
Index: src/main/webapp/resources/javascript/lane-search-facets.js
===================================================================
--- src/main/webapp/resources/javascript/lane-search-facets.js	(revision 104454)
+++ src/main/webapp/resources/javascript/lane-search-facets.js	(working copy)
@@ -1,9 +1,8 @@
 (function() {
     var elt = Y.one('#searchFacets'),
-        searchIndicator = new LANE.SearchIndicator(),
+        searchIndicator = Y.lane.SearchIndicator,
         model = Y.lane.Model,
         basePath = model.get(model.BASE_PATH) || "",
-        query = model.get(model.QUERY),
         encodedQuery = model.get(model.URL_ENCODED_QUERY),
         container = Y.one('#searchResults'),
         facets, i, type, source, Result;
Index: pom.xml
===================================================================
--- pom.xml	(revision 104454)
+++ pom.xml	(working copy)
@@ -215,7 +215,7 @@
                                 <include>lane-ie-min.js</include>
                                 <include>lane-mobile-ad-min.js</include>
                                 <include>lane-persistent-login-min.js</include>
-                                <include>lane-search-indicator-min.js</include>
+                                <include>search-indicator-min.js</include>
                                 <include>lane-bassett-min.js</include>
                                 <include>lane-textinputs-min.js</include>
                                 <include>suggest-min.js</include>
Index: src/main/webapp/resources/javascript/lane-metasearch.js
===================================================================
--- src/main/webapp/resources/javascript/lane-metasearch.js	(revision 104454)
+++ src/main/webapp/resources/javascript/lane-metasearch.js	(working copy)
@@ -3,14 +3,14 @@
         model = Y.lane.Model,
         basePath = model.get(model.BASE_PATH) || "",
         query = model.get(model.QUERY),
-        encodedQuery = model.get(model.URL_ENCODED_QUERY);
+        encodedQuery = model.get(model.URL_ENCODED_QUERY),
+        searchIndicator = Y.lane.SearchIndicator;
     LANE.metasearch = function() {
         var searchElms, // the elements in need of hit counts
             searchables = [], // all engines to search
             searchRequests = [], // search timerIds so we can abort sleeping getResultCounts
             uberEngines = ['cro_', 'mdc_'], // engines with multiple resources
             startTime,
-            searchIndicator = new LANE.SearchIndicator(),
             getSearchUrl = function() {
                 var add, i, y, searchUrl = basePath + '/apps/search/json?q=' + encodedQuery;
                 for (y = 0; y < searchables.length; y++) {
@@ -116,14 +116,14 @@
     if (Y.all('.metasearch').size() > 0 && query) {
         LANE.metasearch.initialize();
         LANE.metasearch.getResultCounts();
-        new LANE.SearchIndicator().show();
+        searchIndicator.show();
     }
     // hybrid search page inputs
     hybridInput = Y.one('.laneSuggest');
     if (hybridInput) {
         laneSuggest = new Y.lane.Suggest(hybridInput);
         laneSuggest.on("select",function(e){
-            new LANE.SearchIndicator().show();
+        	searchIndicator.show();
             hybridInput.ancestor("form").submit();
         });
     }
Index: src/main/webapp/resources/templates/debug.xml
===================================================================
--- src/main/webapp/resources/templates/debug.xml	(revision 104454)
+++ src/main/webapp/resources/templates/debug.xml	(working copy)
@@ -45,8 +45,7 @@
                                         >Textbooks Search</option>
                                 </select>
                                 <span id="searchTips"><a href="/help/choose-search.html">choose your search</a> &#187;</span>
-                                <div id="searchIndicator"><img alt="[searching]"
-                                        src="/resources/images/search-indicator.gif"/></div>
+                                <div id="searchIndicator"></div>
                                 <input name="q" id="searchTerms" type="text"/>
                                 <input id="searchSubmit" type="submit" value="search"/>
                                 <a id="searchReset" title="Clear Search" href="/">×</a>
Index: src/main/webapp/resources/javascript/search-indicator.js
===================================================================
--- src/main/webapp/resources/javascript/search-indicator.js	(revision 0)
+++ src/main/webapp/resources/javascript/search-indicator.js	(working copy)
@@ -0,0 +1,27 @@
+(function() {
+	
+	var SearchIndicator = Y.namespace("lane.SearchIndicator"),
+	    node = Y.one("#searchIndicator");
+	
+	/**
+	 * Adds class "on" to #searchIndicator
+	 * @method show
+	 * @static
+	 */
+	SearchIndicator.show = function() {
+		if (node) {
+			node.addClass("on");
+		}
+	};
+	/**
+	 * Removes class "on" from #searchIndicator
+	 * @method hide
+	 * @static
+	 */	
+	SearchIndicator.hide = function() {
+		if (node) {
+			node.removeClass("on");
+		}
+	};
+	
+})();
\ No newline at end of file
Index: src/main/webapp/resources/javascript/lane-search-indicator.js
===================================================================
--- src/main/webapp/resources/javascript/lane-search-indicator.js	(revision 104454)
+++ src/main/webapp/resources/javascript/lane-search-indicator.js	(working copy)
@@ -1,26 +0,0 @@
-(function() {
-	var model = Y.lane.Model,
-	    basePath = model.get(model.BASE_PATH) || "";
-    LANE.SearchIndicator = function() {
-        var indicator = Y.one("#searchIndicator");
-        if (!indicator) {
-            indicator = Y.Node.create("<div id='searchIndicator'><img src='" + basePath + "/resources/images/search-indicator.gif'/></div>");
-            indicator.setStyle("display","none");
-            Y.one("body").append(indicator);
-        }
-        return {
-            show: function() {
-                indicator.setStyle("display", "block");
-                // IE requires a kick-start to animate indicator gif after form submitted
-                if (Y.UA.ie){
-                    setTimeout(function(){
-                        indicator.one('img').set('src',indicator.one('img').get('src'));
-                    },100);
-                }
-            },
-            hide: function() {
-                indicator.setStyle("display", "none");
-            }
-        };
-    };
-})();
